client/src/hooks/user.ts

```diff
@@ -72,11 +72,27 @@ const useGetUserFromToken = (token: string | null) => {
 //     isError,
 //   };
 // };
+
+const useGoogleLogin = () => {
+  const urlAuthLogin = `${import.meta.env.VITE_BACKEND_AUTH_URL}/oauth2/authorize?response_type=code&client_id=${import.meta.env.VITE_CLIENT_ID}&redirect_uri=${import.meta.env.VITE_REDIRECT_URI}&scope=openid`;
+
+  const handleGoogleLogin = (e: React.MouseEvent<HTMLAnchorElement>) => {
+    e.preventDefault();
+    if (sessionStorage.getItem("isGoogleLogin") === null) {
```

Comment:
This flag ensure that if auth server redirect to /login with google login button is clicked, it will automatically trigger the urlGoogleLogin. Consider this block of code in login screen:

  if (sessionStorage.getItem("isGoogleLogin")) {
    sessionStorage.removeItem("isGoogleLogin")
    window.location.href = urlGoogleLogin
    return null
  }
  
  Btw, I get rid of hasRedirected flag